modules/user/user.service.js

```diff
@@ -35,11 +35,24 @@ class UserService {
       throw new UserInputError('Errors', { errors });
     }
 
-    const checkedUser = await checkUserExist('email', email);
+    const user = await checkUserExist('_id', auth.userId);
+
+    if (auth.email !== email) {
+      const checkedUser = await User.findOne({ email });
+
+      if (checkedUser) {
+        const massage = 'User with provided email already exists';
+        throw new UserInputError(massage, {
+          errors: {
+            email: massage,
+          },
+        });
+      }
+    }
 
     const encryptedPassword = await bcrypt.hash(password, 12);
 
-    const credentials = checkedUser.credentials.map(cred => {
+    const credentials = user.credentials.map(cred => {
```

Comment:
Please, remove logic of updating the password from `updateUser` method. Changing the password should be handled by separate endpoint by providing `oldPassword` and `newPassword`

Reply:
Please, add separate mutation that is responsible for the changing password for the user where it requires providing an old password and new one in separate merge request